python/ray/tune/trainable/function_trainable.py

```diff
@@ -267,41 +266,17 @@ def _start(self):
     def report(self, metrics: Dict, *, checkpoint: Optional[Checkpoint] = None) -> None:
         # TODO(xwjiang): Tons of optimizations.
         self._air_session_has_reported = True
-
-        # TODO(justinvyu): With a unified session, we'll still run into this doubled
-        # report problem. This should be fixed by checking if the checkpoint has been
-        # uploaded already (via some marker), then skipping the repeat upload.
-        if _use_storage_context():
-            assert isinstance(checkpoint, NewCheckpoint)
-            logger.debug(f"Checkpoint received by the Tune session: {checkpoint}")
-            self._fresh_checkpoint = True
-            # TODO(justinvyu): `metrics` doesn't include the autofilled metrics
-            # like `training_iteration` and `time_total_s`.
-            # Should the session be the source of truth for these metrics?
-            self._latest_checkpoint_result = _TrainingResult(
-                checkpoint=checkpoint, metrics=metrics
-            )
-
-            self._last_checkpoint = None
```

Comment:
Note: Reverted this code back to what it was before, since we don't use `_StatusReporter` anymore in the new path.